### PR TITLE
Workflow loader

### PR DESCRIPTION
**Pull request by ronpal**
# Description

Please describe the change you have made.

## Checklist

- [ ] Tests added/updated.
- [ ] Run Demo Job Locally.
- [ ] Documentation updated.
- [ ] Changelogs updated in [CHANGELOG.cdf-tk.md](https://github.com/cognitedata/cdf-project-templates/blob/main/CHANGELOG.cdf-tk.md).
- [ ] Template changelogs updated in [CHANGELOG.templates.md](https://github.com/cognitedata/cdf-project-templates/blob/main/CHANGELOG.templates.md).
- [ ] Version bumped.
  [_version.py](https://github.com/cognitedata/cdf-project-templates/blob/main/cognite/cognite_toolkit/_version.py) and
  [pyproject.toml](https://github.com/cognitedata/cdf-project-templates/blob/main/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite_toolkit/_cdf_tk/load/_resource_loaders.py [hunk 1]
```diff
@@ -1899,3 +1909,114 @@ def count(self, ids: SequenceNotStr[NodeId]) -> int:
     def drop_data(self, ids: SequenceNotStr[NodeId]) -> int:
         # Nodes will be deleted in .delete call.
         return 0
+
+
+@final
+class WorkflowLoader(ResourceLoader[str, WorkflowUpsert, Workflow, WorkflowUpsertList, WorkflowList]):
+    api_name = "workflows"
+    folder_name = "workflows"
+    filename_pattern = r"^(?!.*\b(version|WorkflowVersion)\b).*"
```
Comment (doctrino) [inline cognite_toolkit/_cdf_tk/load/_resource_loaders.py:1918]: This is surprising to me. I expected based on our previous discussion that this should check for files that ends with `.Workflow`? and not files that does not end with `WorkflowVersion` or `version`. Any reason why this changed?

```suggestion
    filename_pattern = r"\.Workflow$"
```

File: cognite_toolkit/_cdf_tk/load/_resource_loaders.py [hunk 2]
```diff
@@ -1899,3 +1909,114 @@ def count(self, ids: SequenceNotStr[NodeId]) -> int:
...
+    def load_resource(
+        self, filepath: Path, ToolGlobals: CDFToolConfig, skip_validation: bool
+    ) -> WorkflowVersionUpsertList:
+        resource = load_yaml_inject_variables(filepath, {})
+
+        workflowversions = [resource] if isinstance(resource, dict) else resource
+        return WorkflowVersionUpsertList.load(workflowversions)
+
+    def retrieve(self, ids: SequenceNotStr[WorkflowVersionId]) -> WorkflowVersionList:
+        return self.client.workflows.versions.list([id for id in ids])
```
Comment (doctrino) [inline cognite_toolkit/_cdf_tk/load/_resource_loaders.py:1997]: Unpacking this as a list does nothing. 
```suggestion
        return self.client.workflows.versions.list(ids)
```

Comment (doctrino) [reply cognite_toolkit/_cdf_tk/load/_resource_loaders.py:1997]: And if `mypy` gives you trouble you can always create a list
```suggestion
        return self.client.workflows.versions.list(list(ids))
```

File: cognite_toolkit/_cdf_tk/load/_resource_loaders.py [hunk 3]
```diff
@@ -1899,3 +1909,114 @@ def count(self, ids: SequenceNotStr[NodeId]) -> int:
...
+        return WorkflowVersionList(updated)
+
+    def delete(self, ids: SequenceNotStr[WorkflowVersionId]) -> int:
+        successes = 0
+        for id in ids:
+            try:
+                self.client.workflows.versions.delete(id)
+                successes += 1
+            except CogniteNotFoundError:
+                print(f"  [bold yellow]WARNING:[/] WorkflowVersion {id} does not exist, skipping delete.")
```
Comment (doctrino) [inline cognite_toolkit/_cdf_tk/load/_resource_loaders.py:2021]: You are only doing a try on the `.delete` call, so lets move the incrementation to `else` block as that is only executed on success. 
```suggestion
        for id in ids:
            try:
                self.client.workflows.versions.delete(id)
            except CogniteNotFoundError:
                print(f"  [bold yellow]WARNING:[/] WorkflowVersion {id} does not exist, skipping delete.")
            else:
                successes += 1
```

File: cognite_toolkit/_cdf_tk/templates/_templates.py [hunk 4]
```diff
@@ -158,6 +158,17 @@ def check_yaml_semantics(parsed: dict | list, filepath_src: Path, filepath_build
         else:
             print(f"      [bold red]:[/] Raw file {filepath_src} has invalid dataformat.")
             exit(1)
+    elif resource_type == "workflows":
```
Comment (doctrino) [inline cognite_toolkit/_cdf_tk/templates/_templates.py:161]: Looking forward until we cleanup this one.

File: tests/tests_unit/test_approval_modules_snapshots/cdf_oid_example_data.yaml [hunk 5]
```diff
@@ -39,7 +39,7 @@ Datapoints:
   - pi_163582
   - pi_163657
   - pi_191092
-  dataframe: 75111115425812004176806621799699857630636183615055067181457655101926656558550
+  dataframe: 83920408757402904477028297651743373327423865323707460346684895807103795112712
```
Comment (doctrino) [inline tests/tests_unit/test_approval_modules_snapshots/cdf_oid_example_data.yaml:42]: This is a mistake that is causing your tests to fail
```suggestion
  dataframe: 75111115425812004176806621799699857630636183615055067181457655101926656558550
```

File: cognite_toolkit/custom_modules/my_example_module/README.md [hunk 6]
```diff
@@ -1,3 +0,0 @@
-# README.md
-
-This is an example of a custom module. It contains two TimeSeries with a dataset.
```
Comment (doctrino) [inline cognite_toolkit/custom_modules/my_example_module/README.md:3]: Any reason why you deleted this module? I guess that has nothing to do with the core change of adding Workflow support?

**Review by doctrino — COMMENTED** [review]
Great job  adding workflow support. The only thing I suggest changing is the regex pattern for the Workflow loader, that is today including any `.yaml` file that does not end with `Version` or `WorkflowVersion`, and instead change it to only `.yaml` files that end with `Workflow`.

File: cognite_toolkit/_cdf_tk/load/_resource_loaders.py (hunk 1 above)
Comment (ronpal) [reply cognite_toolkit/_cdf_tk/load/_resource_loaders.py:1918]: No good reason, just remains of old code.

File: cognite_toolkit/custom_modules/my_example_module/README.md (hunk 6 above)
Comment (ronpal) [reply cognite_toolkit/custom_modules/my_example_module/README.md]: Nope, got a bit eager. Reverted.

**Review by doctrino — APPROVED** [review]
Great :)

**Pull request merged**